### DISCUSSION TITLE
Component Object Model (COM) and Objeck

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#321] [General]
What do you think about COM support for Objeck?

**Comment by objeck** [comment]
Is COM still being used? Last I checked, it is a Windows-only technology.

**Reply by ghost** [reply]
I think it's still being used. I have read an article about automating Windows applications using COM.

**Comment by objeck** [comment]
It's a **massive** legacy framework that competed with CORBA in the late '90s. I found this [thread](https://www.reddit.com/r/cpp/comments/cec8hk/is_com_still_a_valid_technology_are_there_native/). 

I have not done any COM, DCOM, or CORBA development since about 1996. CORBA was added to Java to compete with .Net in 1998 and later removed.

**Reply by ghost** [reply]
Using Objeck to write COM code that can be consumed by other languages (COM Server in Objeck) is too far-fetched. What about the ability to consume COM objects in Objeck (aka COM Client)? You could automate Microsoft Office from Objeck with this.

p/s: COM Server and COM Client are very different from the modern concept of Server and Client.
